pkg/controller/machine_test.go

```diff
@@ -0,0 +1,167 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-test/deep"
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/instance"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+)
+
+type fakeInstance struct {
+	name      string
+	id        string
+	addresses []string
+}
+
+func (i *fakeInstance) Name() string {
+	return i.name
+}
+
+func (i *fakeInstance) ID() string {
+	return i.id
+}
+
+func (i *fakeInstance) Addresses() []string {
+	return i.addresses
+}
+
+func getTestNode(id, provider string) corev1.Node {
+	providerID := ""
+	if provider != "" {
+		providerID = fmt.Sprintf("%s:///%s", provider, id)
+	}
+	return corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("node%s", id),
+		},
+		Spec: corev1.NodeSpec{
+			ProviderID: providerID,
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: fmt.Sprintf("192.168.1.%s", id),
+				},
+				{
+					Type:    corev1.NodeExternalIP,
+					Address: fmt.Sprintf("172.16.1.%s", id),
+				},
+			},
+		},
+	}
+}
+
+var (
+	node1 = getTestNode("1", "aws")
+	node2 = getTestNode("2", "openstack")
+	node3 = getTestNode("3", "")
+
+	nodeList = &corev1.NodeList{
+		Items: []corev1.Node{
+			node1,
+			node2,
+			node3,
+		},
+	}
+)
```

Comment:
Correct me if I'm wrong, but this can be inlined into the `TestController_GetNode` func.